ODIN_II/SRC/netlist_statistic.cpp

```diff
@@ -69,6 +69,42 @@ void init_stat(netlist_t* netlist) {
     netlist->num_logic_element = 0;
 }
 
+int calculate_multiplier_aware_critical_path(nnode_t* node, netlist_t* netlist) {
+    int i, j;
+    int result = 0;
+    if (node->traverse_visited != MULT_OPTIMIZATION_TRAVERSE_VALUE) {
+        /*this is a new node so depth visit it */
+        int backup_value = node->traverse_visited;
+        /* mark that we have visitied this node now */
+        node->traverse_visited = MULT_OPTIMIZATION_TRAVERSE_VALUE;
+        result = 1;
+        for (i = 0; i < node->num_output_pins; i++) {
+            if (node->output_pins[i]->net) {
+                nnet_t* next_net = node->output_pins[i]->net;
+                if (next_net->fanout_pins) {
+                    for (j = 0; j < next_net->num_fanout_pins; j++) {
+                        if (next_net->fanout_pins[j]) {
+                            if (next_net->fanout_pins[j]->node) {
+                                /* recursive call point */
+                                int ans = calculate_multiplier_aware_critical_path(next_net->fanout_pins[j]->node, netlist);
+                                if (result < ans) {
+                                    result = ans;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        result = result + 1;
+        if (node->type == MULTIPLY)
+            result = result * 5;
```

Comment:
this needs to be backed by something... 5 makes no sense unless you give it meaning, maybe at least have a constant somewhere. idk but this is just odd. we've talked about it and theres nothing that makes 5 better than 6, and 6 better than 7 and 7 better than 5... so why 5?

Reply:
@jeanlego I remember that discussion. This is an artifact of copying, I believe there was a branch that was fixing that,but that's no excuse. Have we settled the last time this constant can be ```#define```'d for a little while and then we can extend it with multiplier width?

Reply:
I'd be okay with the width, I think it makes more sense. 

I would look for fanout since you are looking for the multiplier impact (loosely(VERY loosely) related to how many nodes depend on it)